types/signing.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	"encoding/binary"
+
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+)
+
+type Domain [32]byte
+type DomainType uint32
+
+var (
+	DomainBuilder Domain
+)
+
+const (
+	DomainTypeBeaconProposer DomainType = 0x00000000
+	DomainTypeAppBuilder     DomainType = 0x00000001
+)
+
+func init() {
+	DomainBuilder = ComputeApplicationDomain(DomainTypeAppBuilder)
+}
+
+type SigningData struct {
+	Root   Root   `ssz-size:"32"`
+	Domain Domain `ssz-size:"32"`
+}
+
+type forkData struct {
+	CurrentVersion        uint32
+	GenesisValidatorsRoot Root `ssz-size:"32"`
+}
+
+type HashTreeRoot interface {
+	HashTreeRoot() ([32]byte, error)
+}
+
+func ComputeDomain(dt DomainType, forkVersion uint32, genesisValidatorsRoot *Root) [32]byte {
+	if genesisValidatorsRoot == nil {
+		var tmp Root
+		genesisValidatorsRoot = &tmp
+	}
+	forkDataRoot, _ := (&forkData{
```

Comment:
I've noticed here the error is neglected, but given this is a mock, I am not concerned

Reply:
Right, I briefly looked at the generated code for HTR here and it didn't _seem_ like it was possible to error.